### ISSUE TITLE
sshfs-win hangs forever when asked to connect to a remote system for the first time without using "StrictHostKeyChecking=no" option

### ISSUE THREAD

**Issue by mhogomchungu** [billziss-gh/sshfs-win#40]
Steps i took to reproduce:

1. I started by having an empty "known_host" file located at "C:\Program Files (x86)\SSHFS-Win\home\<user>\.ssh"

2. Attempted to create to a remove system with command:
```
C:\Program Files (x86)\SSHFS-Win\bin>sshfs.exe -f ink@192.168.43.195:/home/ink Z:  -o rw,fsname=sshfs@"ink@192.168.43.195:/home/ink",subtype=sshfs,idmap=user,password_stdin
<Entered password>
```
and sshfs hanged forever and the host file was not updated.

3. Running below command worked and the host file was silently updated(notive the addition of StrictHostKeyChecking=no option)
```
C:\Program Files (x86)\SSHFS-Win\bin>sshfs.exe -f ink@192.168.43.195:/home/ink Z:  -o rw,fsname=sshfs@"ink@192.168.43.195:/home/ink",subtype=sshfs,idmap=user,password_stdin,StrictHostKeyChecking=no
<Entered password>
The service sshfs has been started.
The service sshfs has been stopped.
```

The command at step two now works as expected since the host_file is properly updated.

**Comment by billziss-gh** [comment]
SSHFS-Win includes a minimal Cygwin environment that uses `\Program Files (x86)\SSHFS-Win` as its root. This environment does not include user home directories, which I believe are required without the `StrictHostKeyChecking=no`.

The lack of home directories is by design. SSHFS-Win was not really supposed to be used through the command-line.

**Comment by mhogomchungu** [comment]
sshfs-win uses the ```known_host``` host file if its present and writable in a user home directory so this is an accident?

Shouldnt sshfs-win then should automatically set ```StrictHostKeyChecking=no``` if it is meant to be used without the host_file.

Next version of SiriKali auto sets[1] the option on the GUI but users have the option to remove it and i am starting to thing i should remove the option and always auto sets it.

[1] https://github.com/mhogomchungu/sirikali/blob/d959c186275dc89c4f8308ee37478c5c088ece71/src/favorites.cpp#L190

**Comment by billziss-gh** [comment]
I no longer remember the exact behavior of SSH with and without `StrictHostKeyChecking`. However my recollection is that:

- If `StrictHostKeyChecking=no` was missing it would always attempt to create the `known_hosts` file and therefore fail if the user's home directory was missing.
- If `StrictHostKeyChecking=no` was set, it would still use the `known_hosts` file if present, but did not require it.

**Comment by billziss-gh** [comment]
> Next version of SiriKali auto sets[1] the option on the GUI but users have the option to remove it and i am starting to thing i should remove the option and always auto sets it.

Yes, my recommendation would be to remove that option from your users.
